pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -160,7 +162,7 @@ func (s *srv) handleDiscover(c echo.Context) error {
 
 var euiHexPattern = regexp.MustCompile("^eui-([a-f0-9A-F]{16})$")
 
-func (s *srv) handleTraffic(c echo.Context) error {
+func (s *srv) handleTraffic(c echo.Context) (retErr error) {
```

Comment:
`err`

Reply:
If I use `err`, the current error will be referenced:
https://github.com/TheThingsNetwork/lorawan-stack/pull/1198#discussion_r315551032

Reply:
Ehh no, that's not true. The problem in the other thread was that you bind the `err` of the defer to the current value at that point. 

```go
	defer func(error) {
		conn.Disconnect(err)
	}(err)
```

Instead you should have done

```go
	defer func() {
		conn.Disconnect(err)
	}()
```